plugins/plugin-testing/che-plugin-testing-ide/src/main/java/org/eclipse/che/plugin/testing/ide/action/RunDebugTestAbstractAction.java

```diff
@@ -34,55 +35,59 @@
 import javax.validation.constraints.NotNull;
 import java.util.List;
 
-import static org.eclipse.che.api.testing.shared.TestExecutionContext.TestType.CURSOR_POSITION;
+import static org.eclipse.che.api.testing.shared.TestExecutionContext.ContextType.CURSOR_POSITION;
 
 /** Abstract action that analyzes active editor and makes run/debug test actions active if some test methods are exist. */
 public abstract class RunDebugTestAbstractAction extends AbstractPerspectiveAction {
-    TextEditor currentEditor;
-
-    private TestExecutionContext.TestType testType;
-    private List<TestPosition>            testPosition;
-    private boolean                       hasTests;
-    private TestServiceClient             client;
-    private DtoFactory                    dtoFactory;
-    private AppContext                    appContext;
-    private NotificationManager           notificationManager;
-
-    RunDebugTestAbstractAction(EventBus eventBus,
-                               TestServiceClient client,
-                               DtoFactory dtoFactory,
-                               AppContext appContext,
-                               NotificationManager notificationManager,
-                               @Nullable List<String> perspectives,
-                               @NotNull String description,
-                               @NotNull String text,
-                               SVGResource icon) {
+    private TextEditor          currentEditor;
+    private List<TestPosition>  testPosition;
+    private TestServiceClient   client;
+    private DtoFactory          dtoFactory;
+    private AppContext          appContext;
+    private NotificationManager notificationManager;
+
+    public boolean isEditorInFocus;
+    public boolean isEnable;
```

Comment:
I just came across this part - is it really necessary to have these variables `public` when they are not constants?